components/cli/pkg/commands/setup_create_gcp.go

```diff
@@ -57,166 +53,169 @@ func createGcp() error {
-	gcpSpinner.SetNewAction("Updating kube config cluster")
-	updateKubeConfig()
+func CreateGcpRuntimeWithObservability() error {
+	gcpBucketName := configureGCPCredentials()
+	ctx := context.Background()
 
-	hcSql, err := google.DefaultClient(ctx, sqladmin.CloudPlatformScope)
+	// Create a GKE client
+	gcpSpinner := util.StartNewSpinner("Creating GKE client")
+	createKubernentesCluster(ctx, gcpSpinner)
+	sqlService, serviceAccountEmailAddress := configureMysqlGcp(ctx, gcpSpinner)
+	configureBucketGcp(ctx, gcpSpinner, gcpBucketName, serviceAccountEmailAddress, sqlService)
+	configureNFSGcp(gcpSpinner, ctx)
 
+	// Deploy cellery runtime
+	gcpSpinner.SetNewAction("Deploying Cellery runtime")
+	err := deployCelleryRuntime(constants.WITH_OBSERVABILITY)
 	if err != nil {
 		gcpSpinner.Stop(false)
-		fmt.Printf("Error creating client: %v", err)
+		return err
 	}
+	gcpSpinner.Stop(true)
+	return nil
+}
 
-	sqlService, err := sqladmin.New(hcSql)
-	if err != nil {
-		gcpSpinner.Stop(false)
-		fmt.Printf("Error creating sql service: %v", err)
-	}
+func CreateCompleteGcpRuntime() error {
+	gcpBucketName := configureGCPCredentials()
+	ctx := context.Background()
 
-	//Create sql instance
-	gcpSpinner.SetNewAction("Creating sql instance")
-	_, err = createSqlInstance(ctx, sqlService, projectName, region, region, constants.GCP_DB_INSTANCE_NAME+uniqueNumber)
+	// Create a GKE client
+	gcpSpinner := util.StartNewSpinner("Creating GKE client")
+	createKubernentesCluster(ctx, gcpSpinner)
+	sqlService, serviceAccountEmailAddress := configureMysqlGcp(ctx, gcpSpinner)
+	configureBucketGcp(ctx, gcpSpinner, gcpBucketName, serviceAccountEmailAddress, sqlService)
+	configureNFSGcp(gcpSpinner, ctx)
+
+	// Deploy cellery runtime
+	gcpSpinner.SetNewAction("Deploying Cellery runtime")
+	err := deployCelleryRuntime(constants.FULL)
 	if err != nil {
 		gcpSpinner.Stop(false)
-		fmt.Printf("Error creating sql instance: %v", err)
+		return err
 	}
+	gcpSpinner.Stop(true)
+	return nil
+}
 
-	sqlIpAddress, serviceAccountEmailAddress := getSqlServieAccount(ctx, sqlService, projectName, constants.GCP_DB_INSTANCE_NAME+uniqueNumber)
-	fmt.Printf("Sql Ip address : %v", sqlIpAddress)
-
-	// Replace username in /global-apim/conf/datasources/master-datasources.xml
-	if err := util.ReplaceInFile(filepath.Join(util.UserHomeDir(), constants.CELLERY_HOME, constants.GCP, constants.ARTIFACTS, constants.K8S_ARTIFACTS, constants.GLOBAL_APIM, constants.CONF, constants.DATA_SOURCES, constants.MASTER_DATA_SOURCES_XML), constants.DATABASE_USERNAME, constants.GCP_SQL_USER_NAME, -1); err != nil {
+func configureNFSGcp(gcpSpinner *util.Spinner, ctx context.Context) {
+	// Create NFS server
+	gcpSpinner.SetNewAction("Creating NFS server")
+	hcNfs, err := google.DefaultClient(ctx, file.CloudPlatformScope)
+	if err != nil {
 		gcpSpinner.Stop(false)
-		fmt.Printf("%v: %v", constants.ERROR_REPLACING_MASTER_DATASOURCES_XML, err)
+		fmt.Printf("Error getting authenticated client: %v", err)
```

Comment:
Shouldn't we return the errors and handle it properly ? In the errors below as well